src/main/java/com/shep/dao/DatabaseDAO.java

```diff
@@ -0,0 +1,136 @@
+            pstmt.setInt(1, userId);
+            pstmt.setString(2, ticketType);
+            pstmt.executeUpdate();
+        }
+    }
+
+    public User getUserById(int id) throws SQLException {
+        String sql = "SELECT * FROM \"User\" WHERE id = ?";
+        try (Connection conn = connect();
+             PreparedStatement pstmt = conn.prepareStatement(sql)) {
+            pstmt.setInt(1, id);
+            ResultSet rs = pstmt.executeQuery();
+            if (rs.next()) {
+                return new User(rs.getInt("id"), rs.getString("name"), rs.getTimestamp("creation_date"));
+            }
+        }
+        return null;
+    }
+
+    public Ticket getTicketById(int id) throws SQLException {
+        String sql = "SELECT * FROM Ticket WHERE id = ?";
+        try (Connection conn = connect();
+             PreparedStatement pstmt = conn.prepareStatement(sql)) {
+            pstmt.setInt(1, id);
+            ResultSet rs = pstmt.executeQuery();
+            if (rs.next()) {
+                return new Ticket(rs.getInt("id"), rs.getInt("user_id"), rs.getString("ticket_type"), rs.getTimestamp("creation_date"));
+            }
+        }
+        return null;
+    }
+
+    public List<Ticket> getTicketsByUserId(int userId) throws SQLException {
+        List<Ticket> tickets = new ArrayList<>();
+        String sql = "SELECT * FROM Ticket WHERE user_id = ?";
+        try (Connection conn = connect();
+             PreparedStatement pstmt = conn.prepareStatement(sql)) {
+            pstmt.setInt(1, userId);
+            ResultSet rs = pstmt.executeQuery();
+            while (rs.next()) {
+                tickets.add(new Ticket(rs.getInt("id"), rs.getInt("user_id"), rs.getString("ticket_type"), rs.getTimestamp("creation_date")));
+            }
+        }
+        return tickets;
+    }
+
+    public void updateTicketType(int ticketId, String ticketType) throws SQLException {
+        String sql = "UPDATE Ticket SET ticket_type = ?::ticket_type WHERE id = ?";
+        try (Connection conn = connect();
+             PreparedStatement pstmt = conn.prepareStatement(sql)) {
+            pstmt.setString(1, ticketType);
+            pstmt.setInt(2, ticketId);
+            pstmt.executeUpdate();
+        }
+    }
+
+    public void deleteUserById(int userId) throws SQLException {
+        String sql = "DELETE FROM \"User\" WHERE id = ?";
+        try (Connection conn = connect()) {
+            conn.setAutoCommit(false);
+            Savepoint savepoint = conn.setSavepoint();
+            try (PreparedStatement pstmt = conn.prepareStatement(sql)) {
+                pstmt.setInt(1, userId);
+                pstmt.executeUpdate();
+                conn.commit();
+            } catch (SQLException e) {
+                conn.rollback(savepoint);
+                throw e;
+            }
+        }
+    }
+
+    public void updateUserAndTicket(int userId, String userName, int ticketId, String ticketType) throws SQLException {
```

Comment:
the logic can be simplified by using cascades (they are added during the table creation)